### PR TITLE
Adding day length factor switch

### PR DESCRIPTION
**Pull request by jenniferholm**
Adding in a user switch in the parameter file that turns day length factor on (1) by default, or off (2) 

### Description:
This pull request is adding in a user specified switch to turn on or off the day length scaling factor, that is applied to the photosynthetic parameters Vcmax25 and jmax25. This is to be able to allow backwards compatibility with previous versions of the calibration branch.
The function daylength is a scaling factor that is based on the formulation in Bauerle et al. (2012), which quantifies the relationship between day length and Jmax.

### Collaborators:
Greg Lemieux

### Expectation of Answer Changes:
The default is to have the day length factor turned on (equal to 1 in the parameter file), and should have the same results as the most recent calibration branch found here (https://github.com/rgknox/fates/tree/unsupported_sp_calibration_api31)

If turned off, this will change values related to photosynthesis, GPP, and downstream results.

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [x] The in-code documentation has been updated with descriptive comments
- [x] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [ ] FATES PASS/FAIL regression tests were run
- [ ] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*E3SM test hash-tag:*  a7f4ecb9dd

*E3SM baseline hash-tag:*  a7f4ecb9dd

*FATES baseline hash-tag:* 42d804b
*Test Output:*

### CODE REVIEW

**Review by glemieux — CHANGES_REQUESTED** [review]
Thanks for getting this in @jenniferholm.  I have one clarifying question that may require a code update.

File: biogeophys/FatesPlantRespPhotosynthMod.F90 [hunk 1]
```diff
@@ -2343,6 +2351,7 @@ subroutine LeafLayerMaintenanceRespiration_Atkin_etal_2017(lnc_top, &
     integer,  intent(in)  :: ft           ! (plant) Functional Type Index
     real(r8), intent(in)  :: nscaler      ! Scale for leaf nitrogen profile
     real(r8), intent(in)  :: veg_tempk    ! vegetation temperature  (degrees K)
+    real(r8), intent(in)  :: dayl_factor  ! daylength scaling factor (0-1)
```
[the remark this answers is not in the document]
Comment (glemieux) [reply biogeophys/FatesPlantRespPhotosynthMod.F90:2350]: Reviewing this with @rgknox, we're pretty sure this has been superseded and is out of date.

**Comment by glemieux** [comment]
Regression testing on derecho is underway.

**Review by glemieux — APPROVED** [review]

**Comment by glemieux** [comment]
Regression testing with the daylength switch turned off (set to `0`) against the baseline `fates-sci.1.72.3_api.34.0.0-ctsm5.1.dev174-default_allom` create using the new default allometry parameters from #1093, shows all expected tests pass b4b.

Testing with the switch on is in progress.

**Comment by glemieux** [comment]
Regression testing with the switch on results in DIFFs across all fates test mods as expected.  The first timestep appears to be affecting photosynthesis and gpp outputs.

This is ready to integrate after #1128

File: biogeophys/FatesPlantRespPhotosynthMod.F90 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenniferholm) [reply biogeophys/FatesPlantRespPhotosynthMod.F90]: @glemieux - correct, we checked with Jessie today and the dayl_factor should not have been included in the Atkin respiration subroutine, and this was put in here by accident.  I also double checked the Atkins paper, and it's not included.  Not sure why this originally got in?

[the remark this answers is not in the document]
Comment (glemieux) [reply biogeophys/FatesPlantRespPhotosynthMod.F90]: Thanks for confirming @jenniferholm

**Comment by jenniferholm** [comment]
Are there checks left to do on this PR?  Jing needs this update in for her simulations, as do I for some updated CNP runs I'm trying to start.  Maybe others do to for the global calibration? Let me know if there is anything I can do to help review.  Thanks!

**Comment by glemieux** [comment]
@jenniferholm nope, this is all good aside from testing.  It's in the queue and we should be able to integrate it by next week.

**Comment by rosiealice** [comment]
Just noting that I need to also use this for the calibration work (to allow running with both daylength and Jessie's updates to the vertical respiration scaling)

**Comment by rgknox** [comment]
testing now

**Comment by rosiealice** [comment]
Thanks @rgknox :)

**Pull request merged**